test/shift.js

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2013-2019 The NATS Authors
```

Comment:
Looks like a new file.

Also does this have an impact on performance? Have we run those tests?

Reply:
I mean the arg processing on all pub calls.

Reply:
Yes indeed that has a performance.
```
PUBLISHER METRICS 
version                  max   min   samples  rate              
-----------------------  ----  ----  -------  ------------------
0.8.10 (string)          286   281   1000000  3,524,850 msgs/sec
1.3.0                    360   329   1000000  2,935,995 msgs/sec
2.0.0                    436   425   1000000  2,326,483 msgs/sec
0.8.10                   573   550   1000000  1,785,395 msgs/sec
1.0.0                    635   617   1000000  1,602,136 msgs/sec
1.0.0_nodelay            655   626   1000000  1,578,781 msgs/sec
0.8.10 - Buffer.alloc()  958   929   1000000  1,066,666 msgs/sec
0.8.10 - new Buffer()    1022  923   1000000  1,048,987 msgs/sec
0.8.6_1024_mcl           1060  999   1000000  986,635 msgs/sec  
0.8.6                    1026  1013  1000000  978,821 msgs/sec  

RECONNECT 
version                  max   min   samples  rate            
-----------------------  ----  ----  -------  ----------------
0.8.6_1024_mcl           2689  2191  1000000  407,479 msgs/sec
0.8.6                    2947  2168  1000000  406,305 msgs/sec
0.8.10                   3154  2321  1000000  380,966 msgs/sec
0.8.10 - Buffer.alloc()  2964  2337  1000000  378,028 msgs/sec
1.3.0                    3123  2611  1000000  362,318 msgs/sec
2.0.0                    3246  3018  1000000  319,437 msgs/sec

PUBLISH+SUBSCRIBER METRICS 
version                  max   min   samples  rate              
-----------------------  ----  ----  -------  ------------------
0.8.10                   930   881   1000000  1,112,594 msgs/sec
0.8.10 - Buffer.alloc()  929   887   1000000  1,104,972 msgs/sec
0.8.6                    916   900   1000000  1,100,715 msgs/sec
1.0.0_nodelay            939   928   1000000  1,073,422 msgs/sec
1.0.0                    944   926   1000000  1,072,271 msgs/sec
0.8.6_1024_mcl           1014  912   1000000  1,069,391 msgs/sec
1.3.0                    973   918   1000000  1,064,509 msgs/sec
2.0.0                    1074  1013  1000000  964,785 msgs/sec  

REQUEST REPLY METRICS 
version                  max  min  samples  lat                 rate            
-----------------------  ---  ---  -------  ------------------  ----------------
1.3.0                    547  542  100000   2.7191666666666667  183,879 msgs/sec
2.0.0                    625  583  100000   3.002222222222222   166,543 msgs/sec
0.8.10                   622  609  100000   3.0665000000000004  163,052 msgs/sec
0.8.10 - Buffer.alloc()  623  610  100000   3.075               162,601 msgs/sec
0.8.6                    759  725  100000                       136,314 msgs/sec
0.8.6_1024_mcl           760  741  100000                       133,511 msgs/sec
1.0.0                    765  745  100000   3.785               132,100 msgs/sec
1.0.0_nodelay            765  750  100000   3.7874999999999996  132,013 msgs/sec

SUBSCRIBER METRICS 
version                  max   min   samples  rate              
-----------------------  ----  ----  -------  ------------------
1.3.0                    687   639   1000000  1,523,229 msgs/sec
2.0.0                    1035  641   1000000  1,383,763 msgs/sec
0.8.10                   1036  887   1000000  1,032,844 msgs/sec
0.8.10 - Buffer.alloc()  1006  945   1000000  1,028,912 msgs/sec
0.8.6_1024_mcl           1071  1046  1000000  945,278 msgs/sec  
0.8.6                    1084  1059  1000000  935,692 msgs/sec 
```

Reply:
What should I be looking at specifically to compare this exact change?